### PR TITLE
Prometheus: add support for HTTPS

### PR DESCRIPTION
**Pull request by mcorbin**
2 new parameters are added to the prometheus configuration to support
https on the Prometheus endpoint.

I tested it locally, it works.

### CODE REVIEW

**Comment by exo-cedric** [comment]
I don't see anything about the CA(s) bundle used to verify the server's certificate; this should be configurable too (sorry if I missed the obvious)

**Comment by exo-cedric** [comment]
> I don't see anything about the CA(s) bundle used to verify the server's certificate; this should be configurable too (sorry if I missed the obvious)

Ha! Sorry, this is server configuration stuff. Still, we should at one point (maybe _step two_) be able to verify _clients_ certificates (against CA and CRL), as part of the global mTLS effort

**Comment by mcorbin** [comment]
Thanks @exo-cedric, I've added support for client certificates.

i've tested it locally, and it works:

```
curl --cacert demoCA/cacert.pem https://127.0.0.1:7653
curl: (35) error:14094412:SSL routines:ssl3_read_bytes:sslv3 alert bad certificate

curl --cacert demoCA/cacert.pem --key server.key --cert server.crt https://127.0.0.1:7653
<works>
```

for CRL, it's not natively support by golang HTTP servers, so we will have to do it manually (maybe in another PR).

**Comment by exo-cedric** [comment]
> for CRL, it's not natively support by golang HTTP servers, so we will have to do it manually (maybe in another PR).

You're right. CRLs and GoLang... I've seen some code that dies this "extraneously" in _etcd_ code base (but as you say: another story)

File: metrics/prometheus.go [hunk 1]
```diff
@@ -42,6 +48,11 @@ func (c *PrometheusConfiguration) UnmarshalYAML(unmarshal func(interface{}) erro
 	if raw.Interval == config.Duration(0) {
 		return errors.Errorf("missing interval value for prometheus configuration")
 	}
+
+	if (raw.CertFile != "" || raw.KeyFile != "" || raw.CacertFile != "") &&
+		(raw.CertFile == "" || raw.KeyFile == "" || raw.CacertFile == "") {
```
Comment (chrisglass) [inline metrics/prometheus.go]: A bit sad, but it works :)

**Pull request merged**